src/librustc_typeck/check/_match.rs

```diff
@@ -29,12 +29,11 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         pat: &'gcx hir::Pat,
         mut expected: Ty<'tcx>,
         mut def_bm: ty::BindingMode,
-        is_arg: bool)
```

Comment:
The comment on `check_pat_walk` is outdated: be helpful to describe `match_discrim_span` for those without context.